draft-ietf-jsonpath-base.md

```diff
@@ -489,35 +493,30 @@ implementation.
 The semantics are that a valid query is executed against a value,
 the *argument*, and produces a list of zero or more nodes of the value.
 
-The query is a sequence of zero or more *selectors*, each of
-which is applied to the result of the previous selector and provides
-input to the next selector.
+The query is a root identifier followed by a sequence of zero or more *__APPENDERS__*, each of
+which is applied to the result of the previous root identifier or __APPENDER__ and provides
+input to the next __APPENDER__.
 These results and inputs take the form of a *nodelist*, i.e., a
 sequence of zero or more nodes.
 
-The nodelist presented to the first selector contains a single node,
+The nodelist resulting from the root identifier contains a single node,
 the argument.
-The nodelist resulting from the last selector is presented as the
+The nodelist resulting from the last __APPENDER__ is presented as the
 result of the query; depending on the specific API, it might be
 presented as an array of the JSON values at the nodes, an array of
 Normalized Paths referencing the nodes, or both — or some other
 representation as desired by the implementation.
 Note that the API must be capable of presenting an empty nodelist as
 the result of the query.
 
-A selector performs its function on each of the nodes in its input
+An __APPENDER__ performs its function on each of the nodes in its input
 nodelist, during such a function execution, such a node is referred to
 as the "current node".  Each of these function executions produces a
 nodelist, which are then concatenated to produce
-the result of the selector. A node may be selected more than once and
-appear that number of times in the nodelist. Duplicate nodes are not removed.
+the result of the __APPENDER__. A node may be selected more than once and
+appears that number of times in the nodelist. Duplicate nodes are not removed.
 
-The processing within a selector may execute nested queries,
-which conform to the semantics defined here.
-Typically, the argument to that query will be the current node of the
-selector or a set of nodes subordinate to that current node.
-
-A syntactically valid selector MUST NOT produce errors.
+A syntactically valid __APPENDER__ MUST NOT produce errors.
```

Comment:
```suggestion
A syntactically valid __APPENDER__ MUST NOT produce errors when executing the query.
```

... is a bit more explicit.

Reply:
Adopted in ef09a71faad2e8263b38a61c8360feea1564e343.